### DISCUSSION TITLE
Upload speed constantly drops to 0

### DISCUSSION THREAD

**Discussion by idkmanwtt** [nicotine-plus/nicotine-plus#2086] [General]
Every so often someone will try to download something from me, only to ragequit after half an hour and block me because my upload speed is horrendous. I have a 200/200 connection but it seems that I can't reach more than 1mb/s uploading speed on anything or anyone, and sometimes it drops to 0 and stays there for a long time. I have port forwarded normally

I don't know if it's a bug or a problem with my ISP but I genuinely can't upload anything with 200mbps upload speed, i don't know what i'm doing wrong

**Comment by slook** [comment]
> I can't reach more than 1mb/s uploading speed on anything or anyone

This doesn't sound that unusual, although slightly faster speeds might be expected sometimes. The speed depends on many factors as it may be limited at the other end.

> sometimes it drops to 0 and stays there for a long time

This suggests a connection issue of some kind. You could try to investigate what errors are encountered by temporarily enabling Log History > Right Click > Log Categories > "[Debug] Connections". If you find anything of relevance then post your log (using pastebin link or suchlike).

**Comment by mathiascode** [comment]
As mentioned on Reddit, this is on Windows 10.

Could you try SoulseekQt with an alternative account, to see if the issue occurs there too?

**Reply by idkmanwtt** [reply]
it was on soulseekqt too, that's why i switched to nicotine+, soulseekqt had plenty of aborted uploads too

i set an upload limit of 500kb/s and it seems to have stabilized, two people downloaded stuff from me fine without the connection dropping to 0, but they downloaded very small files so i don't know, it's usually only when they try to get a bigger file (more than 100mb) that the issues were starting

**Reply by mathiascode** [reply]
What speeds do you get on speed test sites like https://fast.com (press Show more info for upload speed)? I wonder if the router or something else along the way is acting as a bottleneck.

The code that handles uploads in Nicotine+ is fairly simple, so I don't think we're doing anything special there.

**Reply by idkmanwtt** [reply]
fast.com shows 200dl/200up and a second test 170dl/190up, latency 114ms
speedtest.net shows 203.72dl/199.71up and ping 9ms, second test 212.97dl/200.08up

**Reply by slook** [reply]
It is common for ISP's to show artificially inflated speeds on speed tests by making deals with the network providers. They achieve the high "peak speed" by automatically and inconsistently throttling certain types of uncommon traffic, P2P being of course the first target in order to favour browser based traffic.

Setting an upload limit is always wise, especially for very fast internet connections such as yours.

A few MiB/s is considered reasonable for most users on the Soulseek network, try setting 1000 KiB/s and then try 2000 KiB/s to see if this satisfies your ISP's backbone Fair Usage Policy.

**Reply by idkmanwtt** [reply]
alright will do! thank you so much for your help
